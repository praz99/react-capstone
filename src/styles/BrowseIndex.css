.browse-index-container {
  padding: 10px;
}

.browse-index-container h3 {
  text-align: center;
}

.list-alphabetical {
  display: flex;
  list-style-type: none;
  justify-content: center;
}

.browser-index-links {
  color: #000;
  text-decoration: none;
  padding: 0 5px;
}
