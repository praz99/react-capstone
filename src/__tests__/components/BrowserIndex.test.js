import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import BrowseIndex from '../../components/BrowseIndex';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'http://localhost:3000/search?f=d',
  }),
}));

describe('BrowseIndex component', () => {
  it('Renders correctly', () => {
    const renderer = new ShallowRenderer();
    const wrapper = renderer.render(<BrowseIndex />);
    expect(wrapper).toMatchSnapshot();
  });
});
