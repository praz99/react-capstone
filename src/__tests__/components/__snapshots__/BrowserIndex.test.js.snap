// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseIndex component Renders correctly 1`] = `
<React.Fragment>
  <div
    className="browse-index-container"
  >
    <h3>
      Search Index
    </h3>
    <ul
      className="list-alphabetical"
    >
      <li>
        <a
          className="browse-index-links"
          href="/search?f=a"
        >
          A
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=b"
        >
          B
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=c"
        >
          C
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=d"
        >
          D
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=e"
        >
          E
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=f"
        >
          F
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=g"
        >
          G
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=h"
        >
          H
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=i"
        >
          I
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=j"
        >
          J
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=k"
        >
          K
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=l"
        >
          L
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=m"
        >
          M
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=n"
        >
          N
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=o"
        >
          O
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=p"
        >
          P
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=q"
        >
          Q
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=r"
        >
          R
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=s"
        >
          S
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=t"
        >
          T
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=u"
        >
          U
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=v"
        >
          V
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=w"
        >
          W
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=x"
        >
          X
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=y"
        >
          Y
        </a>
      </li>
      <li>
        <a
          className="browse-index-links"
          href="/search?f=z"
        >
          Z
        </a>
      </li>
    </ul>
  </div>
  <Memo(Connect(DrinkList))
    search="a"
  />
</React.Fragment>
`;
